routes/hotelRoutes.js

```diff
@@ -0,0 +1,301 @@
+const express = require('express');
+const routes = express.Router();
+
+const error = require('../utils/error');
+const { models } = require('../models/index');
+const validateObjectId = require('../middleware/validateObjectId');
+const validateHotelPost = require('../middleware/validateHotelPost');
+const formatHotelPost = require('../middleware/formatHotelPost');
+const validateHotelChange = require('../middleware/validateHotelChange');
+const formatHotelChange = require('../middleware/formatHotelChange');
+
+// ========== HOTEL - created when a Super Admin USER type is created ==========
+
+/**
+ *  @api {get} api/hotel Add new hotel
+ *  @apiVersion 0.1.0
+ *  @apiName postHotel
+ *  @apiGroup Hotels
+ *
+ *  @apiParam {json} hotel hotel information
+ *  @apiParamExample {json} Request-Example:
+ *    {
+ *      "name": "nicolas group",
+ *      "motto": "Function-based contextually-based collaboration",
+ *    }
+ *
+ *  @apiSuccess {String} _id The id of the hotel
+ *  @apiSuccess {String} name The hotel name
+ *  @apiSuccess {String} motto The hotel motto
+ *  @apiSuccess {String} rooms The array of rooms
+ *
+ *  @apiSuccessExample Success-Response: add hotel
+ *    HTTP/1.1 200 OK
+ *    {
+    "_id": "5cc7448e8372e2234f04325f",
+    "name": "Nicholas Group",
+    "motto": "Function-based contextually-based collaboration",
+    "rooms": [],
+    "__v": 0
+ *    }
+ *
+ *  @apiErrorExample Error-Response: minimum hotel requirements not met
+ *    HTTP/1.1 400 BAD REQUEST
+ *    {
+ *      "message": "a hotel must be added with at least a name and motto"
+ *    }
+ */
+routes.post('/', validateHotelPost, formatHotelPost, async (req, res, next) => {
+  const hotel = req.body;
+  try {
+    const newHotel = new models.Hotel(hotel);
+    await newHotel.save();
+    if (newHotel._id) {
+      res.status(201).json(newHotel);
+    } else {
+      res.status(400).json(error.addHotel);
```

Comment:
Will this also returns error message, if the `hotel` already exist in DB?
If not, please add this check.

Reply:
Nice catch. I have added this now but it is only by name - cases aligned to how they will be formatted in the db.